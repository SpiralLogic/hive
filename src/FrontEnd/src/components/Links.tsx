import { FunctionComponent, h } from 'preact';

type Props = { setShowHelp: (value: boolean) => void };
const Links: FunctionComponent<Props> = (props) => {
  const { setShowHelp } = props;
  const handleOnClick = () => {
    setShowHelp(false);
    return false;
  };
  return (
    <div class="links">
      <a class="help" href="#" name="Show rules" onClick={handleOnClick} title="Rules">
        ?
      </a>
      <a class="github" href="https://github.com/SpiralLogic/hive" title="Visit the source code">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100">
          <use href="#github" />
        </svg>
      </a>
    </div>
  );
};

Links.displayName = 'Links';
export default Links;
