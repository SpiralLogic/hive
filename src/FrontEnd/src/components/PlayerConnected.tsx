import '../css/share.css';
import { FunctionComponent, h } from 'preact';
import Modal from './Modal';
type Props = {
  type: 'connected' | 'disconnected';
  setPlayerConnected: (value: 'connected' | 'disconnected' | undefined) => void;
};
const PlayerConnected: FunctionComponent<Props> = (props) => {
  const { type, setPlayerConnected } = props;
  return (
    <Modal name="player connected" onClose={() => setPlayerConnected(undefined)}>
      <p>Player has {type}!</p>
      <button title="New Game" onClick={() => setPlayerConnected(undefined)}>
        Close
      </button>
    </Modal>
  );
};
PlayerConnected.displayName = 'PlayerConnected';
export default PlayerConnected;
