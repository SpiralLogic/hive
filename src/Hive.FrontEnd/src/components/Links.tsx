import '../css/links.css';

import { FunctionComponent, h } from 'preact';
import { useState } from 'preact/hooks';

import { PlayerId } from '../domain';
import { AiAction, HiveEvent } from '../services';
import { getHiveDispatcher, useHiveDispatchListener } from '../utilities/dispatcher';
import { getShareUrl } from '../utilities/share';
import SVG from './SVG';

type Properties = {
  onShowRules: (value: boolean) => void;
  onShowShare: () => void;
  currentPlayer: PlayerId;
};
export const handle = (handler: () => void) => (event: MouseEvent) => {
  event.preventDefault();
  event.stopPropagation();
  handler();
  return false;
};

const Links: FunctionComponent<Properties> = (properties) => {
  const [useAi, setUseAi] = useState(properties.currentPlayer === 0);
  const hiveDispatcher = getHiveDispatcher();
  const clickHandler = () => {
    hiveDispatcher.dispatch<AiAction>({ type: 'toggleAi', newState: !useAi });
    setUseAi(!useAi);
  };

  useHiveDispatchListener<HiveEvent>('opponentConnected', () => {
    setUseAi(false);
  });

  return (
    <div class="links">
      <a
        href={getShareUrl(properties.currentPlayer)}
        name="Share game to opponent"
        title="Share"
        onClick={handle(properties.onShowShare)}>
        <SVG>
          <use href="#share" />
        </SVG>
      </a>
      <a href={`/`} name="New game!" title="New Game">
        <SVG>
          <use href="#new" />
        </SVG>
      </a>
      <a href="#" name="Show rules" onClick={() => properties.onShowRules(true)} title="Rules">
        ?
      </a>
      <a
        href="#"
        name="Toggle Ai"
        class={useAi ? undefined : 'ai-off'}
        onClick={handle(clickHandler)}
        title="Toggle Ai">
        <SVG>
          <use href="#ai" />
        </SVG>
      </a>
      <a class="github" href="https://github.com/SpiralLogic/hive" title="Source code">
        <SVG>
          <use href="#github" />
        </SVG>
      </a>
    </div>
  );
};

Links.displayName = 'Links';
export default Links;
