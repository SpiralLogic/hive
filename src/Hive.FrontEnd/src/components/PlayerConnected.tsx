import '../css/share.css';

import { FunctionComponent } from 'preact';

type Properties = { connected: 'connected' | 'disconnected' | false };
const PlayerConnected: FunctionComponent<Properties> = (properties) => {
  const { connected } = properties;

  return (
    <>
      <p>Player has {connected}!</p>
    </>
  );
};
PlayerConnected.displayName = 'PlayerConnected';
export default PlayerConnected;
