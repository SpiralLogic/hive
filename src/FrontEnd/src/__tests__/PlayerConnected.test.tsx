import { h } from 'preact';
import PlayerConnected from '../components/PlayerConnected';
import { renderElement } from './helpers';
import Links from '../components/Links';

describe('Player connected snapshot tests', () => {
  test('snapshot', () => {
    expect(renderElement(<PlayerConnected />)).toMatchSnapshot();
  });
});
