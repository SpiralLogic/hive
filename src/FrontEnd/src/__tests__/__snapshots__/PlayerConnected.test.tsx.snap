// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player connected snapshot tests snapshot 1`] = `null`;
